### PR TITLE
fix  logOperation multiple read with missing line

### PR DESCRIPTION
**Pull request by huangzhir**
### _Why are the changes needed?_

to close https://github.com/apache/kyuubi/issues/4897

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala [hunk 1]
```diff
@@ -133,8 +133,10 @@ class OperationLog(path: Path) {
       var line: String = reader.readLine()
       while ((i < lastRows || maxRows <= 0) && line != null) {
         logs.add(line)
-        line = reader.readLine()
         i += 1
+        if (i < lastRows || maxRows <= 0) {
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala:137]: thanks for catch this issue!

is `do { ... } while` pattern more suitable here?

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala:137]:
```
    val logs = new JArrayList[String]
    var rows = 0
    try {
      var line: String = null
      do {
        line = reader.readLine()
        if (line != null) { logs.add(line); rows += 1 }
      } while ((rows < lastRows || maxRows <= 0) && line != null)
      (logs, rows)
    } catch {
      ...
    }
```

Comment (huangzhir) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala:137]: than you, this is better solution

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Test failure is not related, thanks, merged to master/1.7

**Pull request closed without merging**